plotly/colors.py

```diff
@@ -339,7 +360,30 @@ def convert_dict_colors_to_same_type(colors, colortype='rgb'):
                                      "for your colortype variable.")
 
 
-def make_colorscale(colors, scale=None):
+def validate_scale_values(scale):
+    """
+    Validates scale values from a colorscale
+    """
+    if len(scale) < 2:
+        raise exceptions.PlotlyError("You must input a list of scale values "
+                                     "that has at least two values.")
+
+    if (scale[0] != 0) or (scale[-1] != 1):
+        raise exceptions.PlotlyError(
+            "The first and last number in your scale must be 0.0 and 1.0 "
+            "respectively."
+        )
+
+    for j in range(1, len(scale)):
```

Comment:
🐄 using `zip`, there are some clear, one-liners that do this check for you http://stackoverflow.com/questions/4983258/python-how-to-check-list-monotonicity